### ISSUE TITLE
`no_std` + `alloc` Boilerplate

### ISSUE THREAD

**Issue by pinkforest** [fission-codes/rust-template#62]
- Mostly the error type 
- `alloc` via feature that `extern crate alloc`
- `heapless` via feature that brings heapless types
- `std` is usually a default feature (when needed) or usually non-default that can be activated for std error type
- error type is making it's way into core though and there is also `core-error` crate

e.g.

- https://github.com/khonsulabs/justjson/commit/968b7860d7263a283f87f104d14109cff33c371a
- https://github.com/khonsulabs/justjson/commit/da7c03cf0719f052d562c0f691c27ed1c320d3d4
- https://github.com/khonsulabs/justjson/commit/330dbd6e1c0aee775a4d527025dca3a381e4f17e
- https://github.com/khonsulabs/justjson/commit/346548569b214138d4177d0e00960a752de27082
- https://github.com/fission-codes/hexxo/commit/982ac3ceb44dc50abd4e7be96c34f75d589b0318

Also thiserror is simple to replace out
- https://github.com/khonsulabs/justjson/commit/70765d9dd2465e20c808479c752b8591061c7da0

**Comment by zeeshanlakhani** [comment]
Yep. A no_std variant with `heapless` would be nice, via prompt.
